mobile/library/common/network/synthetic_address_impl.h

```diff
@@ -55,6 +55,10 @@ class SyntheticAddressImpl : public Instance {
   }
 
 private:
+  void abslHashValue(absl::HashState state) const override {
```

Comment:
Every instance of SyntheticAddressImpl will hash to the same value, even though operator== considers them all unequal. This violates the requirements of absl::Hash and might lead to inefficiencies if these get inserted into absl containers.

I can't tell whether this function will actually be used. If so, it would be safest to follow the rules. Maybe replace `address_` with `this`?

Reply:
I wasn't sure this will be used or not. If it won't be used I can put a `not implement` panic here. @alyssawilk may know this.

Reply:
please no panics in production code.  I don't think it's is used today but it's the sort of thing that I could imagine causing a runtime crash in the data plane if it became used so I'd encourage doing what Todd suggested and maybe adding an ENVOY_BUG if @jpsim or @Augustyniak are confident it should be unused.

Reply:
Got it, learned that, thanks